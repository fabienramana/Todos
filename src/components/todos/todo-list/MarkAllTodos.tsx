import { useState } from "react";

type MarkAllTodosProps = {
    changeStatusOfAllTodos: (status: boolean) => void
}

export default function MarkAllTodos({changeStatusOfAllTodos}: MarkAllTodosProps){

    const [markAllTodosBool, setMarkAllTodosBool] = useState<boolean>(true)

    function updateInputValue(){
        changeStatusOfAllTodos(markAllTodosBool);
        setMarkAllTodosBool(!markAllTodosBool)
    }
    function handleChange(){}

    return (
        <>
            <input id="toggle-all"
            className="toggle-all"
            type="checkbox"
            checked={markAllTodosBool}
            onChange={handleChange}/>
            <label aria-label="mark-all" onClick={() => updateInputValue()}>Mark all as complete</label>
        </>
   )
}