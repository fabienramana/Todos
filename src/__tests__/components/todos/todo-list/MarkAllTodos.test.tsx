import { render, screen, fireEvent } from "@testing-library/react"
import MarkAllTodos from "../../../../components/todos/todo-list/MarkAllTodos"

describe("TodosPage should", ()=> {
    test("call changeStatusOfAllTodos with true if label is clicked", () => {
        const changeStatusOfAllTodos = jest.fn()

        render(<MarkAllTodos changeStatusOfAllTodos={changeStatusOfAllTodos} />)

        const markAllLabel = screen.getByLabelText('mark-all')
  
        fireEvent.click(markAllLabel);

        expect(changeStatusOfAllTodos).toBeCalledTimes(1)
        expect(changeStatusOfAllTodos).toHaveBeenCalledWith(true)
        
    })

    test("call changeStatusOfAllTodos with false if label is clicked for the second time", () => {
        const changeStatusOfAllTodos = jest.fn()

        render(<MarkAllTodos changeStatusOfAllTodos={changeStatusOfAllTodos} />)

        const markAllLabel = screen.getByLabelText('mark-all')
  
        fireEvent.click(markAllLabel);
        fireEvent.click(markAllLabel);

        expect(changeStatusOfAllTodos).toBeCalledTimes(2)
        expect(changeStatusOfAllTodos).toHaveBeenCalledWith(true)
        
    })
    
    test('match snapshot', () => {
  
        const changeStatusOfAllTodos = () => {}
    
        const {asFragment} = render(<MarkAllTodos changeStatusOfAllTodos={changeStatusOfAllTodos} />)
    
        expect(asFragment()).toMatchSnapshot();
      })
})