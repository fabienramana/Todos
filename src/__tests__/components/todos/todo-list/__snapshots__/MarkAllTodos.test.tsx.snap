// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodosPage should match snapshot 1`] = `
<DocumentFragment>
  <input
    checked=""
    class="toggle-all"
    id="toggle-all"
    type="checkbox"
  />
  <label
    aria-label="mark-all"
  >
    Mark all as complete
  </label>
</DocumentFragment>
`;
